src/main/java/io/cdap/plugin/gcp/common/GCPErrorDetailsProvider.java

```diff
@@ -68,18 +69,21 @@ public ProgramFailureException getExceptionDetails(Exception e, ErrorContext err
       if (t instanceof IllegalStateException) {
         return getProgramFailureException((IllegalStateException) t, errorContext);
       }
+      if (t instanceof UnexpectedFormatException) {
+        return getProgramFailureException((UnexpectedFormatException) t, errorContext);
+      }
     }
     return null;
   }
 
   /**
    * Get a ProgramFailureException with the given error
-   * information from {@link IllegalArgumentException}.
+   * information from {@link IllegalArgumentException}, {@link UnexpectedFormatException}.
    *
-   * @param e The IllegalArgumentException to get the error information from.
+   * @param e The IllegalArgumentException or UnexpectedFormatException to get the error information from.
    * @return A ProgramFailureException with the given error information.
    */
-  private ProgramFailureException getProgramFailureException(IllegalArgumentException e,
+  private ProgramFailureException getProgramFailureException(Exception e,
```

Comment:
I think we should combine this and below method, only difference is errorType which can be passed as input param.